File: admin-web/src/api/indicator.ts
import http from './http'
import type { PageData } from './admin'

export interface IndicatorItem {
  id: number
  code: string
  name: string
  level: string
  category_code: string
  category_name: string
  indicator_type: string
  unit: string
  value_range: string
  options: string[]
  data_source: string
  is_feature: boolean
  risk_meaning: string
  weight_star: number
  region: string
  is_veto: boolean
  cycle: string
  scoring_rule: string
  scoring_config: Record<string, any> | null
  display_order: number
}

export interface IndicatorStats {
  total: number
  by_level: Record<string, number>
  by_type: Record<string, number>
  feature: number
  veto: number
  categories: number
}

export interface IndicatorQuery {
  page: number
  size: number
  keyword?: string
  level?: string
  categoryCode?: string
  indicatorType?: string
  isFeature?: boolean
  isVeto?: boolean
}

export function listIndicators(params: IndicatorQuery): Promise<PageData<IndicatorItem>> {
  return http.get('/admin/indicators', { params })
}

export function getIndicatorStats(): Promise<IndicatorStats> {
  return http.get('/admin/indicators/stats')
}

export function getIndicator(code: string): Promise<IndicatorItem> {
  return http.get(`/admin/indicators/${encodeURIComponent(code)}`)
}

export function updateIndicator(code: string, data: Record<string, any>): Promise<IndicatorItem> {
  return http.put(`/admin/indicators/${encodeURIComponent(code)}`, data)
}

export function getIndicatorTree(): Promise<any> {
  return http.get('/indicators/tree')
}

File: app/schemas/indicator.py
"""动态指标体系相关 Pydantic 契约（前端动态表单与指标树）。"""

from typing import Literal

from pydantic import BaseModel


class IndicatorField(BaseModel):
    """动态表单字段配置（对应 indicator_config 14 列 + 派生选项）。"""

    code: str
    name: str
    level: str  # 基本项 / 大类 / 中类 / 小类
    category_code: str
    category_name: str
    indicator_type: Literal["数值", "枚举", "布尔", "文本"] = "数值"
    unit: str = ""
    value_range: str = ""
    options: list[str] = []  # 枚举类型解析出的选项
    data_source: str = ""
    is_feature: bool = False
    risk_meaning: str = ""
    weight_star: float = 3.0
    region: str = ""
    is_veto: bool = False
    cycle: str = ""
    scoring_rule: str = ""
    required: bool = True  # 文本备注类默认非必填


class CategoryNode(BaseModel):
    """指标类别树节点（大类/中类/小类）。"""

    code: str
    name: str
    level: str
    display: str  # "0111 谷物种植"
    indicator_count: int = 0
    children: list["CategoryNode"] = []


class IndicatorTree(BaseModel):
    """指标树：基本项字段 + 类别树。"""

    basic: list[IndicatorField] = []
    categories: list[CategoryNode] = []


class IndicatorConfigOut(BaseModel):
    """渐进式表单配置：基本项 + 选中分支指标 + 当前选择路径。"""

    basic: list[IndicatorField] = []
    indicators: list[IndicatorField] = []
    selected: dict = {}  # {"businessType": "01", "middleType": "011", "smallType": "0111"}
